MigrationScript/Migration.ps1

```diff
@@ -144,7 +181,194 @@ function Main
+					hostname = $($providerProperties.hanaHostname)
+					dbName = $($providerProperties.hanaDbName)
+					sqlPort = $($providerProperties.hanaDbSqlPort).ToString()
+					dbUsername = $($providerProperties.hanaDbUsername)
+					dbPassword = $($providerProperties.hanaDbPassword)
+				}
+			}
+		}
+	}
+
+	# check provider status before making a put call
+	$getResponse = GetAmsV2ProviderStatus -subscriptionId $subscriptionId -resourceGroup $resourceGroupName -monitorName $monitorName -providerName $providerName -logger $logger;
+	$provisioningState = $getResponse.provisiongState;
+	$logger.LogInfo("Current Provisioning State : $provisioningState")
+
+	if($provisioningState -eq "Succeeded") {
+		$logger.LogInfo("Provider $providerName already exists..");
+		Continue;
+	}
+	elseif ($provisioningState -eq "Failed") {
+		$managedKvName = GetAmsV2ManagedKv -subscriptionId $subscriptionId -resourceGroup $resourceGroupName -monitorName $monitorName -logger $logger;
+		$funcName = GetAmsV2ManagedFunc -subscriptionId $subscriptionId -resourceGroup $resourceGroupName -monitorName $monitorName -providerType $providerType -logger $logger;
+		# TODO : code to generate the key name 
+		# DeleteAndPurgeSecretFromKeyVault -keyVaultName "dummyName" -secretKey "dummyKey"
+	}
+
+	# call the put provider method.
+	PutAmsV2Provider -subscriptionId $subscriptionId -resourceGroup $resourceGroupName -monitorName $monitorName -request $requestObj -logger $logger;
+		
+	# we will check the provisioning status for the provider 15 times in 20 sec intervals.
+	$checks = 0;
+
+	# default providioning state is accepted, we will keep checking till is changes.
+	$provisioningState = "Accepted";
+		
+	while ($checks -le 15 -and $provisioningState -eq "Accepted") {
+		Start-Sleep -s 20
+		$getResponse = GetAmsV2ProviderStatus -subscriptionId $subscriptionId -resourceGroup $resourceGroupName -monitorName $monitorName -providerName $providerName -logger $logger;
+		$provisioningState = $getResponse.provisiongState;
+		$checks += 1;
+		$logger.LogInfo("Current Provisioning State : $provisioningState")
+		$logger.LogInfo("Checked the status of Put Provider Call ($checks/15) times.")
+	}
+
+	if($provisioningState -eq "Succeeded") {
+		$logger.LogInfo("Provider $providerName created successfully..");
+	}
+	elseif($provisioningState -eq "Failed") {
+		$logger.LogInfo("Provider $providerName creation failed..");
+	}
+
+	return @{
+		provisiongState = $provisioningState
+	}
+}
+
+<#
+.SYNOPSIS
+Function to migrate the Hana Providers to ams v2
+
+.PARAMETER secretName
+Secret Name, this contains the name of the provider. 
+
+.PARAMETER secretValue
+Secret Value, this contains the provider configuration.
+
+.PARAMETER hostfile
+hostfile object.
+
+.PARAMETER logger
+logger object.
+
+.EXAMPLE
+MigrateNetWeaverProvider -secretName $secret.name -secretValue $secret -logger $logger
+#>
+function MigrateNetWeaverProvider([string]$secretName, $secretValue, $hostfile, $logger) {
```

Comment:
This function is too large, consider refactoring to break it down to smaller , 3-4 functions with better names, that kind of also shows the intent of the code better than any comment.

Reply:
Some refactoring is done. We can take more refactoring as exercise for next PR. Resolving the comment for now. Noted refactoring in upcoming PRs